libraw/bindings.py

```diff
@@ -2,21 +2,124 @@
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 The :class:`libraw.bindings` module handles linking against the LibRaw binary.
-It does not provide an API.
 """
 
 from ctypes import *  # noqa
 from ctypes import util
 
+from libraw import errors
+from libraw.errors import c_error
 from libraw.structs import libraw_data_t, libraw_processed_image_t
 
-# TODO: This is necessary because Travis CI is still using Ubuntu 12.04
-try:
-    # TODO: This will do bad things if the API version isn't 10
-    libraw = cdll.LoadLibrary(util.find_library('raw'))    # pragma: no cover
-    libraw.libraw_init.restype = POINTER(libraw_data_t)    # pragma: no cover
-    libraw.libraw_dcraw_make_mem_image.restype = POINTER(  # pragme: no cover
-        libraw_processed_image_t                           # pragma: no cover
-    )                                                      # pragma: no cover
-except:                                                    # pragma: no cover
-    libraw = cdll.LoadLibrary('')                          # pragma: no cover
+
+class LibRaw(CDLL):
+
+    """
+    A :class:`ctypes.CDLL` that links against `libraw.so` (or the equivalent on
+    your platform).
+    """
+
+    @staticmethod
+    def check_call(exit_code, func, arguments):
+        """
+        Throws a Python error which corresponds to the given LibRaw exit code.
+
+        :param exit_code: the exit code returned by a LibRaw function
+        :type exit_code: :class:`int`
+        :returns: Returns :param:`exit_code` or throws an error from
+                  :class:`libraw.errors`
+        :rtype: :class:`type(exit_code)`
+        """
+
+        if func.restype is c_error and exit_code.value != 0:
+            raise {
+                -1: errors.UnspecifiedError,
+                -2: errors.FileUnsupported,
+                -3: errors.RequestForNonexistentImage,
+                -4: errors.OutOfOrderCall,
+                -5: errors.NoThumbnail,
+                -6: errors.UnsupportedThumbnail,
+                -7: errors.InputClosed,
+                -100007: errors.InsufficientMemory,
+                -100008: errors.DataError,
+                -100009: errors.IOError,
+                -100010: errors.CancelledByCallback,
+                -100011: errors.BadCrop
+            }[exit_code.value]
+
+        return exit_code
+
+    def __init__(self):  # pragma: no cover
```

Comment:
I think __init__is usually the first method. Not sure if it's an actual convention but I've always seen it like that.